elastic/kibana
# [Console] Migrate output editor to Monaco

Closes https://github.com/elastic/kibana/issues/178941

## Summary

This PR migrates the output editor from Ace editor to Monaco editor.

<img width="1491" alt="Screenshot 2024-04-03 at 11 22 27" src="https://github.com/elastic/kibana/assets/59341489/fbff226d-8f6b-470a-b6c8-34f4aaf64200">




**How to test:**
1. Create a `config/kibana.dev.yml` file (if one doesn't exist already) and add the line: `console.dev.enableMonaco: true`
2. Change the condition in https://github.com/elastic/kibana/blob/b8c13babf1c24a2a50668bc5ed647ce87deec5f0/src/plugins/console/public/application/containers/editor/editor.tsx#L76 to `!isMonacoEnabled` to display the old Ace editor in the request panel for now so that we can test sending a request.
3. Send a request and verify the output panel displays the response correctly in json format and that the highlighting is the same as in the original output panel. Also test sending multiple requests at once.

<details>
<summary>JSON output screenshots:</summary>

Ace editor:
<img width="561" alt="Screenshot 2024-03-22 at 16 12 48" src="https://github.com/elastic/kibana/assets/59341489/898a3e48-5a3b-48b6-ac4e-92fa6e05403a">

Now: 
<img width="722" alt="Screenshot 2024-04-03 at 11 12 59" src="https://github.com/elastic/kibana/assets/59341489/503614db-946b-490e-9447-fd3cdcb83bcf">

When multiple requests are sent:
<img width="1491" alt="Screenshot 2024-04-03 at 11 10 53" src="https://github.com/elastic/kibana/assets/59341489/63b05227-a723-49ec-a9ac-3ae3269effd1">

</details>

4. Send a request with `?format=yaml` parameter (e.g. `GET _all?format=yaml`) and verify that the output panel displays the response correctly in yaml format.

<details>
<summary>YAML output screenshot:</summary>
<img width="1459" alt="Screenshot 2024-03-27 at 11 29 45" src="https://github.com/elastic/kibana/assets/59341489/c1967f0f-021e-4b68-bb8f-3098ef35ff1f">

</details>

5. The output also supports text output data. To test this format, delete line 29 and add the following code in line 34 in `src/plugins/console/public/application/containers/editor/monaco/monaco_editor_output.tsx`:

```
const data = [
      {
        response: { value: 'Hello World!' },
      },
    ];
```

Then reload the Console page and verify the data is correctly displayed and highlighted.

<details>
<summary>TEXT output screenshot:</summary>
<img width="285" alt="Screenshot 2024-03-22 at 18 06 31" src="https://github.com/elastic/kibana/assets/59341489/f30cdfd6-be67-4d34-b54d-2a19d6dcb09d">

</details>
